node/resolvers/catalog/index.ts

```diff
@@ -67,13 +67,36 @@ export const queries = {
 
   products: async (_, data, {vtex: ioContext}: ColossusContext, info) => {
     const queryTerm = data.query
+    const isAvailable = data.available
     if (test(/[\?\&\[\]\=\,]/, queryTerm)) {
-      throw new ResolverError(`The query term: '${queryTerm}' contains invalid characters.`, 500)
+      throw new ResolverError(
+        `The query term: '${queryTerm}' contains invalid characters.`,
+        500
+      )
     }
     const url = paths.products(ioContext.account, data)
-    const {data: products} = await axios.get(url, { headers: withAuthToken()(ioContext) })
+    const { data: products } = await axios.get(url, {
+      headers: withAuthToken()(ioContext),
+    })
+    let productsFiltered = products
+
+    if (isAvailable) {
```

Comment:
Extract to separated well-documented function.

Reply:
Where should this function be located?

Reply:
The same file, just extract to a helper function and document.

Reply:
👍